edx_when/api.py

```diff
@@ -398,11 +398,29 @@ def get_schedules_with_due_date(course_id, assignment_date):
     # Using those relative dates, get all Schedules that have a "hit" by working backwards to the start_date
     rel_start_dates = [assignment_date - rel_date for rel_date in rel_dates]
 
-    # Exclude any user that has an overridden date for a course on the specified day so there aren't duplicates
-    return Schedule.objects.filter(
-        enrollment__course_id=course_id,
-        start_date__date__in=rel_start_dates,
-    ).exclude(enrollment__user_id__in=user_ids).select_related('enrollment') | user_date_overridden_schedules
+    if rel_start_dates:
+        # Exclude any user that has an overridden date for a course on the specified day so there aren't duplicates
+        schedules = Schedule.objects.filter(
+            enrollment__course_id=course_id,
+            enrollment__is_active=True,
+            start_date__date__in=rel_start_dates,
+        ).exclude(enrollment__user_id__in=user_ids).select_related('enrollment') | schedules
+
+    # Add in all users with relative dates to exclude from the absolute dates query to prevent duplicates
+    user_ids = schedules.all().values_list('enrollment__user_id', flat=True).distinct()
+
+    has_abs_date_on_day = models.ContentDate.objects.filter(
+        course_id=course_id,
+        policy__abs_date__date=assignment_date
+    ).first()
```

Comment:
ContentDate has an `active` field you might need to check here.

Reply:
Also, you can do `.exists()` which might technically be faster?

Reply:
👍